### PR TITLE
add custom metrics

### PR DESCRIPTION
**Pull request by uwefreidank**
For analyzing issues like exceeding the rate limit on Cloud Foundry, this pull-requests adds some metrics for the HTTP transport layer of the CF clients used by this operator.

The added metrics are:
- `http_client_request_duration_seconds` (histogram)
- `http_client_requests_in_flight` (gauge)
- `http_client_requests_total` (counter)

The new functionality is accompanied by new tests in `client_test.go`.

To ease local testing, it is now possible to enable/disable the webhooks. The default is still enabled.

An example output for the metrics endpoint looks like this:

~~~
# HELP http_client_request_duration_seconds A histogram of HTTP request latencies.
# TYPE http_client_request_duration_seconds histogram
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="0.005"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="0.01"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="0.025"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="0.05"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="0.1"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="0.25"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="0.5"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="1"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="2.5"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="5"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="10"} 1
http_client_request_duration_seconds_bucket{host="http://127.0.0.1:45307",target="cf-api",le="+Inf"} 1
http_client_request_duration_seconds_sum{host="http://127.0.0.1:45307",target="cf-api"} 0.00026072
http_client_request_duration_seconds_count{host="http://127.0.0.1:45307",target="cf-api"} 1
# HELP http_client_requests_in_flight The number in-flight requests for corresponding HTTP client
# TYPE http_client_requests_in_flight gauge
http_client_requests_in_flight{host="http://127.0.0.1:45307",target="cf-api"} 0
# HELP http_client_requests_total The number of HTTP requests from corresponding HTTP client
# TYPE http_client_requests_total counter
http_client_requests_total{code="200",host="http://127.0.0.1:45307",method="get",target="cf-api"} 1
~~~

### CODE REVIEW

**Review by RalfHammer — APPROVED** [review]

File: .vscode/launch.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
                 "--kubeconfig=${workspaceFolder}/.kubeconfig",
                 "--webhook-bind-address=:2443",
                 "--webhook-tls-directory=${workspaceFolder}/.local/ssl",
+                "--enableWebhooks=false",
```
Comment (erichorwath) [inline .vscode/launch.json:17]: disabling webhooks will break the controller since it uses mutating webhooks...

Comment (uwefreidank) [reply .vscode/launch.json:17]: that's why the webhooks are enabled by default, but can be easily disabled for local development.

File: main.go [hunk 2]
```diff
@@ -113,16 +116,18 @@ func main() {
...
 			BindAddress: metricsAddr,
 		},
 		HealthProbeBindAddress: probeAddr,
-	})
+	}
+	if enableWebhooks {
+		options.WebhookServer = webhook.NewServer(webhook.Options{
+			Host: webhookHost,
+			Port: webhookPort,
+		})
```
Comment (erichorwath) [inline main.go:128]: `webhookCertDir` is missing

**Review by RalfHammer — DISMISSED** [review]

**Review by santiago-ventura — APPROVED** [review]

**Review by RalfHammer — APPROVED** [review]

**Review by TheBigElmo — APPROVED** [review]
lgtm

**Pull request merged**